### PR TITLE
Bump wasmparser to 0.85.0

### PR DESCRIPTION
**Pull request by bnjbvr**

### CODE REVIEW

File: symbolic-debuginfo/src/wasm/parser.rs [hunk 1]
```diff
@@ -130,26 +130,24 @@ impl<'data> super::WasmObject<'data> {
...
                         // see https://github.com/WebAssembly/tool-conventions/issues/133
                         "build_id" => {
-                            build_id = Some(data);
+                            build_id = Some(reader.data());
                         }
                         // All of the dwarf debug sections (.debug_frame, .debug_info etc) start with a `.`, and
                         // are the only ones we need for walking the debug info
                         debug if debug.starts_with('.') => {
-                            dwarf_sections.push((name, data));
+                            dwarf_sections.push((reader.name(), reader.data()));
```
Comment (Swatinem) [inline symbolic-debuginfo/src/wasm/parser.rs:143]:
```suggestion
                            dwarf_sections.push((debug, reader.data()));
```

The `debug` binding should be a `&str` I guess?

**Review by Swatinem — APPROVED** [review]

File: symbolic-debuginfo/src/wasm/parser.rs (hunk 1 above)
Comment (bnjbvr) [reply symbolic-debuginfo/src/wasm/parser.rs:143]: Indeed, good catch, thanks!

**Comment by mstange** [comment]
This update caused symbolic's MSRV to increase to 1.58, due to format strings:

```
   Compiling wasmparser v0.85.0
     Running `rustc --crate-name h2 --edition=2018 /builds/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.3.13/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C metadata=c07b053115373f91 -C extra-filename=-c07b053115373f91 --out-dir /builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps --target x86_64-apple-darwin -L dependency=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps -L dependency=/builds/worker/fetches/dump_syms/target/release/deps --extern bytes=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libbytes-a4b23a77b0d5639d.rmeta --extern fnv=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libfnv-f506945c14519358.rmeta --extern futures_core=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libfutures_core-4ae7d23cd314e184.rmeta --extern futures_sink=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libfutures_sink-4356175d269ba6f3.rmeta --extern futures_util=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libfutures_util-180a405b8e489ac3.rmeta --extern http=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libhttp-8fc2ea9c09aed8dc.rmeta --extern indexmap=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libindexmap-960a8110a8ca7502.rmeta --extern slab=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libslab-45b91366405779c5.rmeta --extern tokio=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libtokio-72d235f02e51c747.rmeta --extern tokio_util=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libtokio_util-2766438102773de9.rmeta --extern tracing=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libtracing-af2e4cf1942fa091.rmeta --cap-lints allow -C linker=/builds/worker/workspace/build/src/taskcluster/scripts/misc/osx-cross-linker`
     Running `rustc --crate-name wasmparser --edition=2021 /builds/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmparser-0.85.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C metadata=33ce9de1a0097475 -C extra-filename=-33ce9de1a0097475 --out-dir /builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps --target x86_64-apple-darwin -L dependency=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps -L dependency=/builds/worker/fetches/dump_syms/target/release/deps --extern indexmap=/builds/worker/fetches/dump_syms/target/x86_64-apple-darwin/release/deps/libindexmap-960a8110a8ca7502.rmeta --cap-lints allow -C linker=/builds/worker/workspace/build/src/taskcluster/scripts/misc/osx-cross-linker`
error: there is no argument named `func_index`
   --> /builds/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmparser-0.85.0/src/validator/component.rs:188:110
    |
188 | ...   format!("lowered parameter types `{:?}` do not match parameter types `{:?}` of core function {func_index}", params.as_slice(), core...
    |                                                                                                    ^^^^^^^^^^^^

error: there is no argument named `func_index`
   --> /builds/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmparser-0.85.0/src/validator/component.rs:195:104
    |
195 | ...   format!("lowered result types `{:?}` do not match result types `{:?}` of core function {func_index}", results.as_slice(), core_ty.r...
    |                                                                                              ^^^^^^^^^^^^

error: there is no argument named `MAX_WASM_TYPE_SIZE`
  --> /builds/worker/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmparser-0.85.0/src/validator.rs:83:63
   |
83 |             format!("effective type size exceeds the limit of {MAX_WASM_TYPE_SIZE}"),
   |                                                               ^^^^^^^^^^^^^^^^^^^^

error: could not compile `wasmparser` due to 3 previous errors
```

**Comment by Swatinem** [comment]
We don’t have an MSRV for symbolic, the readme even states (https://github.com/getsentry/symbolic#building-and-development):

> To build the Rust crate, we require the latest stable Rust, as well as a C++11 compiler.

I think thats even a C++14 compiler by now due to Swift demangling.

**Comment by mstange** [comment]
Ah, I see. That's fair.

**Pull request merged**